### PR TITLE
[posix] network: split into platform specific files

### PR DESCRIPTION
**Pull request by lrusak**
This is just a bit of a code shuffle to split the posix network code into the relevant platforms. Much of this code is old and can probably be improved but is better left to the platform maintainers to do so.

These new platform abstractions inherit from `CNetworkPosix` which may not be the best idea to have a common parent class but it doesn't harm for now. If a class needs to remove the inheritance in the future they are free to do so. This doesn't add or change any existing functionality simply shuffles code to allow for easier future improvement.

### CODE REVIEW

File: xbmc/platform/darwin/osx/network/NetworkOsx.cpp [hunk 1]
```diff
@@ -21,6 +22,11 @@
 #include <netinet/if_ether.h>
 #include <sys/sockio.h>
 
+namespace
+{
+#define ARRAY_SIZE(X) (sizeof(X) / sizeof((X)[0]))
```
Comment (Rechi) [inline xbmc/platform/darwin/osx/network/NetworkOsx.cpp]: putting a define into a namespace doesn't have any effect

File: cmake/treedata/freebsd/subdirs.txt [hunk 2]
```diff
@@ -3,8 +3,9 @@ xbmc/cores/VideoPlayer/Process/rbpi cores/VideoPlayer/Process/rbpi
 xbmc/input/touch                    input/touch
 xbmc/input/touch/generic            input/touch/generic
 xbmc/platform/freebsd               platform/freebsd
+xbmc/platform/freebsd/network       platform/freebsd/network
 xbmc/platform/linux/input           platform/linux/input
-xbmc/platform/linux/network         platform/linux/network
+xbmc/platform/linux/network/zeroconf platform/linux/network/zeroconf
```
Comment (Rechi) [inline cmake/treedata/freebsd/subdirs.txt]: Move this line to `cmake/treedata/optional/common/avahi.txt` and add ` # AVAHI` at the end of the line.

Comment (lrusak) [reply cmake/treedata/freebsd/subdirs.txt]: good idea :+1:

File: xbmc/platform/linux/network/zeroconf/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+if(AVAHI_FOUND)
```
Comment (Rechi) [inline xbmc/platform/linux/network/zeroconf/CMakeLists.txt]: There is no need for this check, as this file is only processed if Avahi is found.

**Comment by lrusak** [comment]
So why are these platforms adding `platform/linux/network/zeroconf` to the build if AVAHI is not found?

**Comment by Rechi** [comment]
Found the reason, the code adding the subdirectories only checks `ENABLE_AVAHI`, but not `FOUND_AVAHI`.
https://github.com/xbmc/xbmc/blob/e48ccc9bb41ab9a34c771a3d4cf3b6aa334a2a18/cmake/scripts/common/Macros.cmake#L512-L517

**Pull request merged**